onmt/translate/translator.py

```diff
@@ -661,27 +571,28 @@ def _translate_batch(
             mb_device = memory_bank.device
         memory_lengths = tile(src_lengths, beam_size)
```

Comment:
Tiling is specific to beam search. I suggest adding an `initialize` method to `DecodeStrategy` to do all these operations (step "(2)") only for beam search.